### ISSUE TITLE
---pdf-engine in Pandoc 2

### ISSUE THREAD

**Issue by iandol** [htdebeer/pandocomatic#38]
Currently pandocomatic assumes latex intermediate format if to=pdf

```ruby
                # There is no "pdf" output format; change it to latex but keep the
                # extension.
                value = "latex" if option == "to" and value == "pdf"
```
https://github.com/htdebeer/pandocomatic/blob/master/lib/pandocomatic/command/convert_file_command.rb#L198

But with Pandoc 2 we now have more flexible options to use several HTML>PDF engine (see https://github.com/htdebeer/paru/pull/28). In Pandoc you specify this simply with the `--output` file name, but pandocomatic does not use this mechanism due to its multiple file processor system, so we need to think of the best way to do this. The simplest is have custom to types like `to=html-pdf` — but this seems a bit klunky. Another option is pandocomatic reads the `pdf-engine` setting and then changes the logic of `value=html|latex` depending on the passed engine? But that means it needs to have parsed `pdf-engine` before `to` so the logic needs to move outside this loop.

```ruby
if option == 'to' and value == 'pdf'
  value='html5' if pdf-engine =~ /wkhtmltopdf|weasyprint|prince/
  value='latex' if pdf-engine =~ /latex/
end
```

**Comment by iandol** [comment]
This was the original issue: https://github.com/jgm/pandoc/issues/3906 — and the pull that closed it: https://github.com/jgm/pandoc/commit/c7e3c1ec1797d633c181e5701c94f4169b0e5471 and the updated manual (not yet HTMLified): https://github.com/jgm/pandoc/blob/master/MANUAL.txt#L185 

Pandoc does not support to="pdf", it only supports setting -o="file.pdf" and `pdflatex` is the default if no pdf-engine is specified. If pdf-engine is invalid and no to= then the default should be HTML I think...

**Comment by htdebeer** [comment]
All right. I'll see what I can do. Thanks for the information!

**Comment by htdebeer** [comment]
According to the manual, LaTeX is the default. So, when not specified, pandocomatic will default to LaTeX as well.

I suppose a 'PDF' base template that sets the engine for a user's system is quite handy if she prefers another engine over the default LaTeX one.

This feature is included in version 0.1.4.17
